stdlib/public/Platform/CMakeLists.txt

```diff
@@ -101,7 +101,8 @@ foreach(sdk ${SWIFT_SDKS})
 
     # If this SDK is a target for a non-native host, create a native modulemap
     # without a sysroot prefix. This is the one we'll install instead.
-    if(NOT "${SWIFT_SDK_${SWIFT_HOST_VARIANT_SDK}_ARCH_${arch}_PATH}" STREQUAL "/")
+    if(NOT "${SWIFT_SDK_${SWIFT_HOST_VARIANT_SDK}_ARCH_${arch}_PATH}" STREQUAL "/" AND
```

Comment:
This seems to be for when the Swift toolchain is cross-compiled for another environment where the C headers are kept in `/`, but that doesn't apply for Android.

Reply:
Correct me if I'm wrong, but this CMake config seems to be geared towards cross-compiling the Swift host toolchain, say for linux ARMv7 from linux x86_64, where you then copy that host toolchain to a linux ARMv7 host and run it there and this stripped glibc.modulemap will pick up the local C headers in /usr. That will never be the case for Android, whether NDK or not, as Android doesn't come with C headers.

This didn't affect me on Termux before, as it uses the prefixed glibc.modulemap in the swift-android-aarch64 build directory. But once I tried `--install-swift` to package the toolchain, it copies over this un-prefixed glibc.modulemap instead, so I noticed this and disabled it.

Reply:
This is related to the stdlib, not the toolchain. At this point the toolchain should already be compiled. This piece of code is run for every stdlib that's being compiled (when building natively, Android will be build because it is the host, when building cross-compiled, it will be build because the user requested so).

This line seems to only deal with the Unix SDKs, and it seems to deal with the problem that when cross-compiling, the path to the headers is `/sysroot/foo/bar`, but when deploying the headers will be in `/foo/bar`. In Termux is weirder, because even if you are compiling natively, the sysroot is technically `/data/.../termux`, and you don't want to remove it when installing. That's why I was proposing to use `SWIFT_ANDROID_NATIVE_SYSROOT`.

Reply:
I am saying that this ANDROID check alone suffices, because the NDK config, ie non-Termux cross-compiles, _also_ will never deploy the Android stdlib and glibc.modulemap to an un-prefixed Android host, since such an Android host with C headers in /usr doesn't exist.

Reply:
I know that Android seemingly distributes a incorrect modulemap with partial paths. But those paths are relative to the sysroot, and are the right paths for the NDK, which is helpful (more helpful than leaking the private full path of the NDK of the build machine). That's why I would prefer not to modify the Android NDK results from what it is now.

In some way, Android should have taken the path that Windows took and force users to link a constant modulemap in the NDK directory, but modifying other project files seems even worse IMO.

Reply:
Alright, you're right that I'm making this more general than I needed, and since you want to ship the un-prefixed modulemap instead when generated with the NDK without a native Android sysroot, I'll change it.